include/deal.II/distributed/full_distributed_tria.h

```diff
@@ -0,0 +1,315 @@
+       * cell.
+       */
+      std::vector<std::pair<unsigned int, types::boundary_id>> boundary_ids;
+    };
+
+
+
+    /**
+     * Data to construct a parallel::fullydistributed::Triangulation.
+     */
+    template <int dim, int spacedim>
+    struct ConstructionData
+    {
+      /**
+       * Cells of the locally-relevant coarse-grid triangulation.
+       */
+      std::vector<CellData<dim>> cells;
+
+      /**
+       * Vertices of the locally-relevant coarse-grid triangulation.
+       */
+      std::vector<Point<spacedim>> vertices;
+
+      /**
+       * Mapping from coarse-grid index to coarse-grid id.
+       */
+      std::vector<types::coarse_cell_id> coarse_cell_index_to_coarse_cell_id;
+
+      /**
+       * CellInfo for each locally relevant cell on each level.
+       */
+      std::vector<std::vector<CellInfo<dim>>> cell_infos;
+    };
+
+
+    /**
+     * A distributed triangulation with a distributed coarse grid.
+     */
+    template <int dim, int spacedim = dim>
+    class Triangulation
+      : public parallel::DistributedTriangulationBase<dim, spacedim>
+    {
+    public:
+      typedef typename dealii::Triangulation<dim, spacedim>::cell_iterator
+        cell_iterator;
+
+      typedef
+        typename dealii::Triangulation<dim, spacedim>::active_cell_iterator
+          active_cell_iterator;
+
+      typedef
+        typename dealii::Triangulation<dim, spacedim>::CellStatus CellStatus;
+
+
+      enum Settings
+      {
+        /**
+         * Default settings, other options are disabled.
+         */
+        default_setting = 0x0,
+        /**
+         * This flags needs to be set to use the geometric multigrid
+         * functionality. This option requires additional computation and
+         * communication. Note: geometric multigrid is still a work in
+         * progress.
+         */
+        construct_multigrid_hierarchy = 0x1
+      };
+
+
+      /**
+       * Constructor.
+       *
+       * @param mpi_communicator denotes the MPI communicator to be used for the
+       *                         triangulation.
+       * @param settings See the description of the Settings enumerator.
+       */
+      explicit Triangulation(MPI_Comm       mpi_communicator,
+                             const Settings settings = default_setting);
+
+      /**
+       * Destructor.
+       */
+      virtual ~Triangulation() = default;
+
+      /**
+       * Create a triangulation from the provided ConstructionData.
+       *
+       * @note The namespace dealii::fullydistributed::Util contains functions
+       *       to create ConstructionData.
+       *
+       * @note This is the function to be used instead of create_triangulation.
+       *
+       * @param construction_data
+       */
+      void
+      reinit(const ConstructionData<dim, spacedim> &construction_data);
+
+      /**
+       * @note Use the function reinit() instead of this function to create
```

Comment:
"This function is not implemented for this class. Instead, use the function reinit() to create the triangulation."